### PR TITLE
Added pnpm

### PR DESCRIPTION
**Pull request by Akkuma**
Locally I could not get npm5 + shrinkwrap working, but I added pnpm @zkochan would be great to do a quick once over to ensure I've implemented everything correctly.

My own results look like:
```
Run 1
npm 4.x: 26.67s (average 26.67s)
npm 4.x (cached): 8.19s (average 8.19s)
npm 4.x (shrinkpacked): 11.93s (average 11.93s)
npm 4.x (shrinkpacked, compressed): 11.67s (average 11.67s)
yarn: 10.23s (average 10.23s)
yarn --offline: 8.24s (average 8.24s)
npm 5.x: 7.81s (average 7.81s)
npm 5.x (cached): 7.01s (average 7.01s)
pnpm: 8.56s (average 8.56s)
pnpm (cached): 4.20s (average 4.20s)
pnpm --offline: 4.32s (average 4.32s)

Run 2
npm 4.x: 28.74s (average 27.71s)
npm 4.x (cached): 8.15s (average 8.17s)
npm 4.x (shrinkpacked): 11.26s (average 11.59s)
npm 4.x (shrinkpacked, compressed): 11.74s (average 11.70s)
yarn: 9.90s (average 10.07s)
yarn --offline: 8.33s (average 8.29s)
npm 5.x: 7.65s (average 7.73s)
npm 5.x (cached): 6.42s (average 6.71s)
pnpm: 8.65s (average 8.61s)
pnpm (cached): 4.27s (average 4.23s)
pnpm --offline: 5.53s (average 4.92s)

Run 3
npm 4.x: 28.55s (average 27.99s)
npm 4.x (cached): 8.21s (average 8.18s)
npm 4.x (shrinkpacked): 11.13s (average 11.44s)
npm 4.x (shrinkpacked, compressed): 15.23s (average 12.88s)
yarn: 13.10s (average 11.08s)
yarn --offline: 11.13s (average 9.23s)
npm 5.x: 9.38s (average 8.28s)
npm 5.x (cached): 7.25s (average 6.89s)
pnpm: 9.74s (average 8.98s)
pnpm (cached): 4.98s (average 4.48s)
pnpm --offline: 5.11s (average 4.99s)

Run 4
npm 4.x: 28.65s (average 28.15s)
npm 4.x (cached): 9.66s (average 8.55s)
npm 4.x (shrinkpacked): 14.60s (average 12.23s)
npm 4.x (shrinkpacked, compressed): 16.21s (average 13.71s)
yarn: 12.88s (average 11.53s)
yarn --offline: 12.80s (average 10.13s)
npm 5.x: 10.81s (average 8.91s)
npm 5.x (cached): 8.38s (average 7.27s)
pnpm: 12.28s (average 9.81s)
pnpm (cached): 6.19s (average 4.91s)
pnpm --offline: 6.17s (average 5.28s)

Run 5
npm 4.x: 30.49s (average 28.62s)
npm 4.x (cached): 8.41s (average 8.52s)
npm 4.x (shrinkpacked): 13.61s (average 12.51s)
npm 4.x (shrinkpacked, compressed): 12.75s (average 13.52s)
yarn: 10.81s (average 11.38s)
yarn --offline: 10.24s (average 10.15s)
npm 5.x: 8.91s (average 8.91s)
npm 5.x (cached): 6.66s (average 7.14s)
pnpm: 11.41s (average 10.13s)
pnpm (cached): 5.89s (average 5.11s)
pnpm --offline: 6.06s (average 5.44s)
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,84 +2,69 @@ const path = require('path');
...
+    runBenchmark('npm', ['install'], 'shrinkpack-compressed');
+    runBenchmark('yarn', ['install'], 'yarn');
+    runBenchmark('yarn', ['install', '--offline'], 'yarn-offline');
+    runBenchmark('npm5', ['install'], 'npm');
+    runBenchmark('npm5', ['install'], 'npm-cached');
+    runBenchmark('npm5', ['install'], 'shrinkpack');
+    runBenchmark('npm5', ['install'], 'shrinkpack-compressed');
+    runBenchmark('pnpm', ['install'], 'pnpm');
+    runBenchmark('pnpm', ['install'], 'pnpm-cached');
+    runBenchmark('pnpm', ['install'], 'pnpm-offline');
```
Comment (zkochan) [inline index.js]: why don't you pass the `--offline` param to pnpm? It won't work in offline mode w/o it.

**Review by zkochan — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (Akkuma) [reply index.js]: I originally had it and removed it, because I thought the npmrc file was enough to force it

Comment (zkochan) [reply index.js]: npmrc should be fine as well. I don't see an npmrc file with `offline = true` in this PR though.

**Review by zkochan — APPROVED** [review]

**Comment by zkochan** [comment]
oh, and pnpm should be added to the top of README, where all the different package managers are listed

**Comment by Akkuma** [comment]
@JamieMason or @siddharthkp do either of you want to run this and update the results, since npm5+shrinkwrap won't run for me?

**Comment by JamieMason** [comment]
Sure @Akkuma, would you mind updating your branch with the latest from master please? Then I'll be happy to do the rest when I'm next free on Thursday eve. Thanks a lot for this.

**Comment by JamieMason** [comment]
I've run out of time for today but fyi your work is merged into https://github.com/JamieMason/npm-cache-benchmark/commits/Akkuma-master and will be released once some other bits have been finished up, thanks.

**Comment by JamieMason** [comment]
I'll close this PR as it'll be going into https://github.com/JamieMason/npm-cache-benchmark, please add any comments at https://github.com/JamieMason/npm-cache-benchmark/issues/4 – thanks.

**Pull request closed without merging**